src/main.ts

```diff
@@ -198,7 +197,7 @@ import {AsyncWorkQueue} from './utils';
   const coordinateBlock = await new Promise((resolve) => {
     let interval = setInterval(() => {
       try {
-        const coordinateBlock = embed!.shadowRoot!
+        const coordinateBlock = redditCanvas!.embed!.shadowRoot!
```

Comment:
Do you actually need a `!` here? You check `!redditCanvas` above, that should've narrowed down the type to not include `null`

Reply:
I think I added this before going back and doing the check, so yeah, no longer needed on a bunch of these.

Reply:
Actually TS's type system can't handle that. As soon as you enter another function it can't track that anymore.

Reply:
What do you mean by enter another function?